File: modul_valoracions_ios/Examples/Test_ObjectiveC/Test_ObjectiveC/ViewController.m
//
//  ViewController.m
//  Test_ObjectiveC
//
//  Created by Antonio García on 29/01/2019.
//  Copyright © 2019 OpenRoad. All rights reserved.
//

#import "ViewController.h"
@import AppStoreRatings;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Update AppStoreRatings status
    NSURL *url = [NSURL URLWithString:@"https://www.openroad.es/projects/appstoreratings/test/ratings_config.json"];
    [AppStoreRatings.shared updateRatingStatsWithConfigUrl:url completion:^(BOOL isDialogRequested, NSError * _Nullable error) {
        if (error == nil) {
            NSLog(@"Finished: isDialogRequested %@", isDialogRequested ? @"yes":@"no");
        }
        else {
            NSLog(@"error %@", error.localizedDescription);
            
            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Error" message:error.localizedDescription preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil];
            [alertController addAction:okAction];
            [self presentViewController:alertController animated:YES completion:nil];
        }
    }];
    
    // Debug internal status
#ifdef DEBUG
    NSLog(@"%@", [AppStoreRatings.shared currentStatusDescription]);
    
    [AppStoreRatings.shared debugCurrentStatusWithConfigURL:url completion:^(BOOL isDialogRequested, NSInteger launchCountsRemaining, double daysRemaining, BOOL wasPreviouslyRequested, NSError * _Nullable error) {
        if (error) {
            NSLog(@"error %@", error.localizedDescription);
        }
        else {
            NSLog(@"isDialogRequested %@", isDialogRequested ? @"yes":@"no");
            NSLog(@"wasPreviouslyRequested %@", wasPreviouslyRequested ? @"yes":@"no");
            NSLog(@"launchCountsRemaining %ld",launchCountsRemaining);
            NSLog(@"daysRemaining %@", [NSString stringWithFormat:@"%.3f", daysRemaining] );
        }
    }];
    
#endif
}


@end

File: modul_valoracions_ios/Examples/Test_Swift/Test_Swift/ViewController.swift
//
//  ViewController.swift
//  Test_Swift
//
//  Created by Antonio García on 25/01/2019.
//  Copyright © 2019 OpenRoad. All rights reserved.
//

import UIKit
import AppStoreRatings

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Update AppStoreRatings status
        let url = URL(string: "https://www.openroad.es/projects/appstoreratings/test/ratings_config.json")!
        AppStoreRatings.shared.updateRatingStats(configURL: url) { result in
            switch result {
            case .success(let isDialogRequested):
                NSLog("Finished: isDialogRequested: \(isDialogRequested)")
                
            case .failure(let error):
                NSLog("error: \(error.localizedDescription)")
                let alertController = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
                let okAction = UIAlertAction(title: "Ok", style: .default, handler: nil)
                alertController.addAction(okAction)
                self.present(alertController, animated: true, completion: nil)
            }
        }
        
        // Debug internal status
        #if DEBUG
        NSLog("status: \(AppStoreRatings.shared.currentStatusDescription())")
        
        NSLog("Library internal status:");
        AppStoreRatings.shared.debugCurrentStatus(configURL: url) { result in
            switch result {
            
            case let .success(willRequestDialog, launchCountsRemaining, daysRemaining, wasPreviouslyRequested):
                NSLog("willRequestDialog: \(willRequestDialog)")
                NSLog("wasPreviouslyRequested: \(wasPreviouslyRequested)")
                NSLog("launchCountsRemaining: \(launchCountsRemaining)")
                NSLog("daysRemaining: \(String(format: "%.3f", daysRemaining))")

            case .failure(let error):
                NSLog("error: \(error.localizedDescription)")
            }
        }
        #endif
    }


}

